components/tidb_query/src/rpn_expr/impl_encryption.rs

```diff
@@ -29,36 +33,103 @@ fn hex_digest(hashtype: MessageDigest, input: &[u8]) -> Result<Bytes> {
 
 #[cfg(test)]
 mod tests {
+    use tipb::ScalarFuncSig;
+
     use super::*;
     use crate::rpn_expr::types::test_util::RpnFnScalarEvaluator;
-    use tipb::ScalarFuncSig;
+
+    fn test_unary_func_ok_none<I: Evaluable, O: Evaluable>(sig: ScalarFuncSig)
+    where
+        O: PartialEq,
+        Option<I>: Into<ScalarValue>,
+        Option<O>: From<ScalarValue>,
+    {
+        assert_eq!(
+            None,
+            RpnFnScalarEvaluator::new()
+                .push_param(Option::<I>::None)
+                .evaluate::<O>(sig)
+                .unwrap()
+        );
+    }
+
+    #[test]
+    fn test_md5() {
+        let test_cases = vec![
+            ("", "d41d8cd98f00b204e9800998ecf8427e"),
+            ("a", "0cc175b9c0f1b6a831c399e269772661"),
+            ("ab", "187ef4436122d1cc2f40dc2b92f0eba0"),
+            ("abc", "900150983cd24fb0d6963f7d28e17f72"),
+            ("123", "202cb962ac59075b964b07152d234b70"),
+            ("你好", "7eca689f0d3389d9dea66ae112e5cfd7"),
+            ("分布式データベース", "63c0354797bd261e2cbf8581147eeeda"),
+        ];
+        for (arg, expect_output) in test_cases {
+            let arg = Some(arg.as_bytes().to_vec());
+            let expect_output = Some(Bytes::from(expect_output));
+
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(arg)
+                .evaluate::<Bytes>(ScalarFuncSig::Md5)
+                .unwrap();
+            assert_eq!(output, expect_output);
+        }
+
+        let invalid_utf8_test_cases = vec![
+            (vec![0xc0, 0x80], "b26555f33aedac7b2684438cc5d4d05e"),
+            (vec![0xED, 0xA0, 0x80], "546d3dc8de10fbf8b448f678a47901e4"),
+        ];
+        for (arg, expect_output) in invalid_utf8_test_cases {
+            assert!(std::str::from_utf8(&arg).is_err());
+            let expect_output = Some(Bytes::from(expect_output));
+
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(arg)
+                .evaluate::<Bytes>(ScalarFuncSig::Md5)
+                .unwrap();
+            assert_eq!(output, expect_output);
+        }
```

Comment:
It's better to merge this logic with the one above. For example:

```rust
 ("a".as_bytes().to_vec(), "0cc175b9c0f1b6a831c399e269772661"),
```

This is because it's much more difficult to find the bugs in test executor than in case list; fewer executor is better.